package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,22 @@
 #!/bin/sh
 
-CFG=$1
+REAL_CFG=$1
 
-[ -n "$CFG" ] || CFG=/etc/board.json
+[ -n "$REAL_CFG" ] || REAL_CFG=/etc/board.json
 
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
+if [ -d "/etc/board.d/" ] && [ ! -s "$REAL_CFG" ]; then
+	# Use temp file to prevent incomplete file on power-cut, CFG is used by the sourced script to read/write the file
+	CFG="$(dirname "$REAL_CFG")/.$(basename "$REAL_CFG").tmp"
```

Comment:
I am not a shell expert, but why not use this?
```suggestion
	CFG="$REAL_CFG.tmp"
```

Reply:
I am fine with either